diff --git a/archiso/initcpio/hooks/archiso b/archiso/initcpio/hooks/archiso
index 853652e..53215eb 100644
--- a/archiso/initcpio/hooks/archiso
+++ b/archiso/initcpio/hooks/archiso
@@ -197,6 +197,39 @@ archiso_mount_handler() {
         fi
     fi
 
+    msg ":: Creating config dir ..."
+    mkdir -p /run/archiso/config/
+
+    msg ":: Execute the configuration processing script ..."
+    jsonconfig="/run/archiso/config/sysrescue-effective-config.json"
+    if ! /usr/bin/sysrescue-configuration.lua > /tmp/sysrescue-configuration.log 2>&1
+    then
+        echo "ERROR: The configuration processing script has failed"
+        cat /tmp/sysrescue-configuration.log
+        sleep 20
+    fi
+
+    msg ":: Define early boot options based on the configuration ..."
+    for curentry in "copytoram" "checksum" "loadsrm"
+    do
+        if value=$(jq --raw-output ".global.${curentry} | tostring | ascii_downcase" ${jsonconfig} 2>/dev/null)
+        then
+            if [[ "${value}" == 'y' ]] || [[ "${value}" == 'yes' ]] || [[ "${value}" == 't' ]] || \
+               [[ "${value}" == 'true' ]] || [[ "${value}" == 'on' ]] || [[ "${value}" == '1' ]] || \
+               [[ "${value}" == '1.0' ]] ; then
+                echo "Entry '.global.${curentry}' enabled in config: value='${value}'"
+                eval "${curentry}='y'"
+            elif [[ "${value}" == 'n' ]] || [[ "${value}" == 'no' ]] || [[ "${value}" == 'f' ]] || \
+                 [[ "${value}" == 'false' ]] || [[ "${value}" == 'off' ]] || [[ "${value}" == '0' ]] || \
+                 [[ "${value}" == '0.0' ]] || [[ "${value}" == 'null' ]] ; then
+                echo "Entry '.global.${curentry}' disabled in config: value='${value}'"
+            else
+                echo "ERROR: Found invalid value for '.global.${curentry}': value='${value}'"
+                sleep 8
+            fi
+        fi
+    done
+
     if [[ "${checksum}" == "y" ]]; then
         if [[ -f "/run/archiso/bootmnt/${archisobasedir}/${arch}/airootfs.sha512" ]]; then
             msg -n ":: Self-test requested, please wait..."
