diff -u -r archiso-43.orig/archiso/initcpio/install/archiso_pxe_common archiso-43/archiso/initcpio/install/archiso_pxe_common
--- archiso-43.orig/archiso/initcpio/install/archiso_pxe_common	2022-12-15 21:48:53.560000001 +0000
+++ archiso-43/archiso/initcpio/install/archiso_pxe_common	2022-12-15 21:51:30.649999999 +0000
@@ -8,13 +8,16 @@
     add_binary /usr/lib/initcpio/ipconfig /bin/ipconfig
 
     # Add hosts support files+dns
-    add_symlink /usr/lib/libnss_files.so.2 $(readlink /usr/lib/libnss_files.so.2)
-    add_binary $(readlink -f /usr/lib/libnss_files.so.2)
-    add_symlink /usr/lib/libnss_dns.so.2 $(readlink /usr/lib/libnss_dns.so.2)
-    add_binary $(readlink -f /usr/lib/libnss_dns.so.2)
+    add_binary /usr/lib/libnss_files.so.2
+    add_binary /usr/lib/libnss_dns.so.2
+    add_binary /usr/lib/libnss_mymachines.so.2
+    add_binary /usr/lib/libnss_resolve.so.2
 
     add_dir /etc
     echo "hosts: files dns" > $BUILDROOT/etc/nsswitch.conf
+
+    # the lua dns library (used in sysrescue-configuration.lua) requires /etc/hosts to exist
+    touch $BUILDROOT/etc/hosts
 }
 
 help() {
