diff --git a/archiso/initcpio/hooks/archiso b/archiso/initcpio/hooks/archiso
index 3eb7ac7..e4585af 100644
--- a/archiso/initcpio/hooks/archiso
+++ b/archiso/initcpio/hooks/archiso
@@ -247,6 +247,14 @@ archiso_mount_handler() {
 
     _mnt_sfs "/run/archiso/bootmnt/${archisobasedir}/${arch}/airootfs.sfs" "/run/archiso/sfs/airootfs"
 
+    if [[ "${copytoram}" == "y" ]] && ls -lh /run/archiso/bootmnt/autorun* >/dev/null 2>/dev/null; then
+        msg -n ":: Copying autorun scripts to RAM..."
+        if ! cp /run/archiso/bootmnt/autorun* /run/archiso/copytoram/ ; then
+            echo "ERROR: while copy '/run/archiso/bootmnt/autorun*' to '/run/archiso/copytoram/'"
+            launch_interactive_shell
+        fi
+    fi
+
     [[ "${loadsrm}" == "y" ]] && _mnt_srm "/run/archiso/bootmnt/${archisobasedir}"
 
     if [[ -f "/run/archiso/sfs/airootfs/airootfs.img" ]]; then
